dace/transformation/interstate/state_elimination.py

```diff
@@ -154,11 +154,146 @@ def apply(self, sdfg):
                     break
             else:
                 # If removed assignment does not appear in any other edge,
-                # remove symbol
+                # replace and remove symbol
+                if assignments_to_consider[varname] in sdfg.symbols:
+                    sdfg.replace(varname, assignments_to_consider[varname])
                 if varname in sdfg.symbols:
                     sdfg.remove_symbol(varname)
 
 
+def _alias_assignments(sdfg, edge):
+    assignments_to_consider = {}
+    for var, assign in edge.assignments.items():
+        if assign in sdfg.symbols or (assign in sdfg.arrays and isinstance(
+                sdfg.arrays[assign], dt.Scalar)):
+            assignments_to_consider[var] = assign
+    return assignments_to_consider
+
+
+@registry.autoregister_params(strict=True)
+class SymbolAliasPromotion(transformation.Transformation):
+    """
+    1. We define inter-state edge assignments to be "SymbolAlias" when a symbol
+    has is set equal to another symbol or (potentially promotable) scalar.
+    2. We define inter-state edge assignment promotion as moving the assignment
+    to the previous inter-state edge according to the topological order, when
+    this is unambiguous.
+    3. SymbolAliasPromotion moves SymbolAlias assignments to the previous
+    inter-state edge according to the topological order.
+
+    The purpose of this transformation is to iteratively move symbolic aliases
+    together, so that true duplicates can be easily removed.
```

Comment:
I cannot understand what this transformation does from this docstring. I would first make it start with the second paragraph ("Iteratively moves..."), and explain what is a symbolic alias.